src/banking_stage.rs

```diff
@@ -91,30 +94,33 @@ impl BankingStage {
         let bank_starting_tx_count = bank.transaction_count();
         let count = mms.iter().map(|x| x.1.len()).sum();
         let proc_start = Instant::now();
-        for (msgs, vers) in mms {
-            let transactions = Self::deserialize_transactions(&msgs.read().unwrap());
-            reqs_len += transactions.len();
-            let transactions = transactions
-                .into_iter()
-                .zip(vers)
-                .filter_map(|(tx, ver)| match tx {
-                    None => None,
-                    Some((tx, _addr)) => if tx.verify_plan() && ver != 0 {
-                        Some(tx)
-                    } else {
-                        None
-                    },
-                })
-                .collect();
+        let transactions: Vec<_> = mms
```

Comment:
@garious
garious 5 days ago  Member
Yeah, I'm not saying the code is incorrect, just that it'd perform best if the focus was on pipelining flat_map calls (filter_map is a specialization of flat_map), so that you can maximize parallelization of deserialize.

I'd think the optimal code would be structured something like this:

let msgs = mms.iter().flat_map(|x| x.0.read().unwrap());
let vers = mms.iter().flat_map(|x| x.1);
let txs = Self::deserialize_transactions(msgs);
let txs = txs.into_iter().zip(vers).filter_map(...);

Reply:
Yes, I am trying the changes suggested by @garious above. Having slight difficulty due to rust knowledge. If we want to land the current change, I can push the modifications separately.

I am making the change to update node count in the test. I'll push that as part of this PR.